mmv1/third_party/terraform/acctest/vcr_utils.go

```diff
@@ -287,6 +345,73 @@ func ReformConfigWithProvider(config, provider string) string {
 	return string(resourceHeader.ReplaceAll(configBytes, providerReplacementBytes))
 }
 
+// reformConfigWithProvider reformats the config to use the given provider
+// The method matches a regex for the provider block and replaces it with the given provider.
+// For example: ' data "google_compute_network" "default" { provider = "google-local" } '
+// will be reformatted to ' data "google_compute_network" "default" { provider = "google-beta" } '
+// parseReleaseDiffOutput reads the temporary file created during the release diff test and returns the last line of output
+// This is useful for extracting the diff output from the file after the test has run
+
+func ParseReleaseDiffOutput(temp *os.File) (string, string, error) {
+	if temp == nil {
+		return "", "", errors.New("temporary file is nil")
+	}
+
+	_, err := temp.Seek(0, io.SeekStart)
+	if err != nil {
+		return "", "", fmt.Errorf("failed to seek to beginning of temporary file: %w", err) // Updated return
+	}
+
+	var lastLine string
+	var allContentBuffer bytes.Buffer
+	scanner := bufio.NewScanner(temp)
+
+	for scanner.Scan() {
+		line := scanner.Text()
+		lastLine = line
+		allContentBuffer.WriteString(line)
+		allContentBuffer.WriteByte('\n')
+	}
+
+	if err := scanner.Err(); err != nil {
+		return "", "", fmt.Errorf("error reading temporary file: %w", err)
+	}
+
+	allContent := allContentBuffer.String()
+
+	if len(allContent) > 0 && allContent[len(allContent)-1] == '\n' {
+		allContent = allContent[:len(allContent)-1]
+	}
+
+	return lastLine, allContent, nil
+}
+
+func writeOutputFileDeferFunction(tempFile *os.File, regularFailureFile *os.File, diffFailureFile *os.File, failed bool) {
+	if tempFile != nil {
+		// parses the temporary file created during the release diff test and returns the last line of output
+		// This is useful for extracting the diff output from the file after the test has run
+
+		var lastLine, output, err = ParseReleaseDiffOutput(tempFile)
```

Comment:
I would consider rearranging the logic a bit for clarity and so you don't need to pass around as many variables. One option would be to first read the temp file to a string, then have an `isDiffStep` function that returns a boolean, then write to the failure file based on that result.